content/docs/plugin-docs/json/presetsv2.md

```diff
@@ -0,0 +1,228 @@
+
+- Text
+- File
+- Selection box
+- Numbers
+- Switch
+
+
+#### Input Types
+
+Input types are "boxes" where the user can input different type of data values like `text`, `image`, `models` and more. Input type is defined in the properties of each input box inside the manifest of a preset. To create a new input box create a new array inside of `fields` array.
+
+
+###### Text
+
+`textInput` is the default type of input is `text`. To create a `text` input type make a array with the name of the text box as a `string` followed by another `string` which is the variable where the user's input will be assigned to
+
+
+###### File
+
+`fileInput` input types can be created by an array this will include the name of the input box, the input type. Last by creating a `object` properties are defined such as `accept`, `icon` these can only be used inside a `fileInput` input type.
+
+
+###### Selection Box
+
+`selectInput` is an input type creates selection boxes or drop down lists that have a fixed amount of options. To make a selection box create a input box with a name and assign a variable to it, then create a properties object and define the type of input box as `selectInput` then a `default` value this is the default option the input box will be set. Create an array named `options` then for each option create an object that will define what the display `text` and `value` should be.
+
+
+###### Number Input
+
+`numberInput` is an input type that creates a slider this slider goes from a specified `min` number to a specified `max` number. This slider will count up or down by a specified `step` number. To create a `numberInput` create a new input box with the `numberInput` as the input `type` then put a `min`, `max`, `step` number.
+
+
+###### Switch
+
+`switch` is the simplest type of input which is a literal switch that can be turned on or off. The variable that is assigned to the input will get `0` or `1` based on the switch state (`0` for `off` and `1` for `on`). To create it set the input `type` to `switch`
+
+
+##### Input Properties
+
+Input Properties are poperties that can be defined inside of an input box these are grouped in a `object` and will change depending on what input type is selected. There are 3 properties that work with all input types and those are:
+
+- `type` Type of input.
+- `default` Default value for the input.
+- `optional` Whether or not the input is optional.
+
+
+### Icon
+Icons show up in the presets window. Presets use [Material Design Icons](https://materialdesignicons.com/) a collection of icons that can be defined in the manifest.json by typing `mdi-` followed by the icon's name.
+
+
+### Variables
+
+Variables are able to be used in the `createFiles` and `expandFile` components of the `manifest.json they` can also be used any other files in the preset as long as they are "injected" in the file.
```

Comment:
"`manifest.json they`" -> "`manifest.json`. They"
"... also be used any other files in the preset ..." -> "...also be used in any other files of the preset..."